roles/storwize/tasks/main.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: /etc/cinder/ssh
+  file: dest=/etc/cinder/ssh state=directory owner=cinder group=cinder
+
+- name: Generate ssh keys
+  shell: ssh-keygen -b 2048 -t rsa -f /etc/cinder/ssh/id_rsa -q -N ""
+  args:
+    creates: /etc/cinder/ssh/id_rsa
+  when: primary_controller
+
+- name: copy ssh key from primary
+  slurp: path=/etc/cinder/ssh/id_rsa
+  run_once: true
+  delegate_to: "{{ groups['controller'][0] }}"
+  register: v7k_ssh_key  
+  when: primary_controller == False
+
+- name: write /etc/cinder/ssh/id_rsa
+  copy:
+    dest: /etc/cinder/ssh/id_rsa
+    content: "{{ v7k_ssh_key['content']  }}"
+    mode: 0600
+  when: primary_controller == False
+
+- name: /etc/cinder/ssh
+  file: dest=/etc/cinder/ssh/id_rsa owner=cinder group=cinder
+
+- name: install sshpass
+  package: name="sshpass"
+  register: result
+  until: result|succeeded
+  retries: 5
+  when: primary_controller
+  
+
+- name: upload public key to server
+  shell: sshpass -p "{{san_password}}" scp /root/.ssh/id_rsa.pub {{san_login}}@{{management_ip}}:/home/admin/upgrade
+  when: primary_controller
+
+- name: add key to user
+  shell:  sshpass -p "{{san_password}}" ssh  {{san_login}}@{{management_ip}} "chuser -keyfile /home/admin/upgrade/id_rsa.pub {{san_login}} "
+  when: primary_controller
+
+- name: update ip configuration
```

Comment:
again this should not be done here. For ursula, it should meet the pre-reqs and configured that way.

If we need to configure ip/etc it should be done in flotsam.